src/sage/rings/number_field/number_field_ideal.py

```diff
@@ -995,16 +995,36 @@ def is_prime(self):
             False
             sage: K.ideal(17).is_prime()  # ramified
             False
+
+        TESTS:
+
+        Check that we do not factor the norm of the ideal, this used
+        to take half an hour, see :trac:`33360`::
+
+            sage: K.<a,b,c> = NumberField([x^2-2,x^2-3,x^2-5])
+            sage: t = (((-2611940*c + 1925290/7653)*b - 1537130/7653*c
+            ....:       + 10130950)*a + (1343014/7653*c - 8349770)*b
+            ....:       + 6477058*c - 2801449990/4002519)
+            sage: t.is_prime()
+            False
         """
         try:
             return self._pari_prime is not None
         except AttributeError:
-            F = self.factor()  # factorization with caching
-            if len(F) != 1 or F[0][1] != 1:
-                self._pari_prime = None
-            else:
-                self._pari_prime = F[0][0]._pari_prime
-            return self._pari_prime is not None
+            pass
+
+        K = self.number_field().pari_nf()
+        I = self.pari_hnf()
+
+        self._pari_prime = K.idealismaximal(I) or None
```

Comment:
Do we have to worry about thread-safety? There's a window of time here were the cache in the ideal thinks it is a prime but it has not been certified yet.

Maybe better to save this to a local variable, then certify the primality, and then fill the object?

Reply:
I don't think Sage generally cares about thread safety, does it?

Reply:
> I don't think Sage generally cares about thread safety, does it?

Why not?

Reply:
I don't know, but I suspect the reason is that CPython's `threading` implementation does not provide multi-core parallelism.